Resources/doc/cookbook/recipe_improve_performance_large_datasets.rst

```diff
@@ -0,0 +1,34 @@
+Improve performance of large datasets
+=====================================
+
+If your database table contains thousands of records, the database queries generated
+by SonataAdmin may become very slow. Here are tips how to improve the performance of your admin.
+
+
+Change default Pager to SimplePager
+-------------------------------------
+
+Default `Pager` is counting all rows in the table, so user can easily navigate
+to any page in the Datagrid. But counting thousands or milions of records
+can be slow operation. If you don't need to know the number of all records,
+you can use `SimplePager` instead. It doesn't count all rows, but gives user only
+information if there is next page or not.
+
+To use SimplePager in your admin just change the `$pagerType` property from
+`default` to `simple`.
+
+.. code-block:: php
+
+    // YourNS/YourBundle/Admin/PostAdmin.php
+
+    ...
+
+    class PostAdmin extends Admin
+    {
+        /**
+         * Pager type
+         *
+         * @var string
+         */
+        protected $pagerType = 'simple';
```

Comment:
It should be also possible to configure this with the main ``sonata_admin`` configuration. The use case is admin from external vendor, it is not possible to create and configure a new class just for altering the pager type.